public_html/wp-content/plugins/multi-event-sponsors/javascript/index.js

```diff
@@ -0,0 +1,177 @@
+/* global MultiEventSponsor */
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { Button, Dashicon, Spinner } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { PluginSidebar } from '@wordpress/edit-post';
+import { createInterpolateElement, useState } from '@wordpress/element';
+import { __, _x, sprintf } from '@wordpress/i18n';
+import { registerPlugin } from '@wordpress/plugins';
+import { decodeEntities } from '@wordpress/html-entities';
+
+const { stripTags } = wp.sanitize;
+
+/*
+ * Render the main view for the sidebar.
+ */
+function PushToActiveCamps( { adminUrl } ) {
+	const [ loading, setLoading ] = useState( false );
+	const [ result, setResult ] = useState( null );
+
+	const sourcePost = useSelect(
+		( select ) => {
+			// This doesn't update until the `Update` button is saved, but that's good enough for our uses.
+			return select( 'core/editor' ).getCurrentPost();
+		}
+	);
+
+	const isDirty = useSelect(
+		( select ) => {
+			return select( 'core/editor' ).isEditedPostDirty();
+		}
+	);
+
+	if ( isDirty && result !== null ) {
+		// Make sure the UI doesn't show results from a previous push once the current changes are saved.
+		setResult( null );
+	}
+
+	return (
+		<PluginSidebar
+			name="push-to-active-camps"
+			className="push-to-active-camps"
+			title="Push to Active Camps"
+			icon={ <Dashicon icon="heart" /> }
+		>
+			<p>
+				{ sprintf(
+					// translators: Title of the post
+					__( 'This will copy the title/content/etc of this post to all of its corresponding %s posts on active WordCamp sites (except regional camps).', 'wordcamporg' ),
```

Comment:
How does it know not to push to regional camps?

Reply:
AFAIK we don't have a way to identify regional camps programmatically. I'm assuming that it doesn't need to, though, since regional camps will either delete the posts after the site is created, or edit them. That's a good assumption to check and document, though.

@harishankerr , is that correct? or do regional camps sometimes leave the global sponsor posts unedited?

Reply:
It also only pulls published posts, so it'd skip ones they moved to draft

Reply:
Okay, so it assumes regional camps have edited posts, not that it does any special exclusion for regional camps? If a regional camp hasn't edited a sponsor, it will push out the update? Overall I think that's fine, I would just update the message to make the difference clear:

> This will copy the title/content/etc of this post to all of its corresponding %s posts on active WordCamp sites, unless they have edited the post.